### DISCUSSION TITLE
Is there Sidebery CSS documentation?

### DISCUSSION THREAD

**Discussion by manyfew** [mbnuqw/sidebery#1895] [Q&A]
I know very little CSS (mostly learnt through trial and errod) and I see a lot of snippets like this:

`.Tab[data-active="true"] .body {`

How do people know they have to type that exact text? Where can I find what I need to target specific parts/elements of Sidebery?

**Comment by soul4kills** [comment]
If there is no documentation available, you'll have to go through the raw *.css files yourself to see what "selectors/classes/custom variables" are used. Once you know what element you want to target, then starts the trial and error.

Usually the selectors/classes/custom variables are named in a way that describes it's relation to the interface pretty well. Not always but usually they are.

It's always going to be trial and error even with proper documentation. Documentation just helps minimize the frustration.

**Reply by manyfew** [reply]
You'll have to forgive my ignorance but what is the CSS file? I can locate it on my drive and take a look. Incidentally, how long would it take a not-too-bright but inquiring mind to learn enough CSS? I just read on Reddit you never stop learning CSS, which is a little discouraging.

**Reply by soul4kills** [reply]
https://github.com/mbnuqw/sidebery/tree/v5/src/styles/themes/proton/sidebar
This is a good spot to start to see what elements you can reference and modify. Just go through the files and look for what's used. The files end in ".styl". Usually CSS is contained in a file that ends in ".css". You'll see once you start getting into it more.

Also check out r/firefoxCSS on reddit. It's a good place to see what others are doing with customizing things with CSS and it'll give you things to try out and play with.

CSS should be easy to pick up quickly. It's a markup language. A lot easier to learn than a programing language.
I personally do not know CSS inside out. But I know enough to use it for whatever I need it for.

https://www.w3schools.com/css/default.asp - Good place to learn, has a demo area where you can play with the CSS and see how it works in real time.

https://developer.mozilla.org/en-US/docs/Web/CSS/display - Another good place to reference and it has great inline examples to show you how things work.

> how long would it take a not-too-bright but inquiring mind to learn enough CSS?

All that's required is the inquiring part, the brights will come later. You'll have fun CSS. You'll have a lot of things to apply it to.
